// wxWindows Patch
// Modified to be compatible with current unreleased library
// 10-24-2001 Robert Cavanaugh
// Description:
// This patch fixes a bug in wxWizard::ShowPage() where if
// the user wishes to change the displayed bitmap in two
// consecutive pages, it does not work. This bug exists
// in the released libraries as well

--- Tip/wxWindows/src/generic/wizard.cpp	Fri Aug 17 11:52:02 2001
+++ New/wxWindows/src/generic/wizard.cpp	Thu Oct 25 10:55:26 2001
@@ -3,5 +3,8 @@
 // Purpose:     generic implementation of wxWizard class
 // Author:      Vadim Zeitlin
-// Modified by:
+// Modified by: Robert Cavanaugh
+//              1) Added capability for wxWizardPage to accept resources
+//              2) Added "Help" button handler stub
+//              3) Fixed ShowPage() bug on displaying bitmaps
 // Created:     15.08.99
 // RCS-ID:      $Id: wizard.cpp,v 1.19 2001/08/17 17:52:02 VZ Exp $
@@ -74,7 +81,20 @@
 // ----------------------------------------------------------------------------
@@ -247,7 +267,17 @@
     bool btnLabelWasNext = TRUE;
 
-    // and this tells us whether we already had the default bitmap before
-    int bmpWasDefault;
+    // Modified 10-20-2001 Robert Cavanaugh.
+    // Fixed bug for displaying a new bitmap
+    // in each *consecutive* page
+
+    // flag to indicate if this page uses a new bitmap
+    bool bmpIsDefault = TRUE;
+
+    // use these labels to determine if we need to change the bitmap
+    // for this page
+    wxBitmap   PreviousBitmap = wxNullBitmap;
+    wxBitmap   ThisBitmap = wxNullBitmap;
 
+    // check for previous page
     if ( m_page )
     {
@@ -264,13 +294,13 @@
 
         btnLabelWasNext = m_page->GetNext() != (wxWizardPage *)NULL;
-        bmpWasDefault = !m_page->GetBitmap().Ok();
-    }
-    else // no previous page
+
+        // Get the bitmap of the previous page (if it exists)
+        if(m_page->GetBitmap().Ok())
     {
-        // always set the bitmap
-        bmpWasDefault = -1;
+            PreviousBitmap = m_page->GetBitmap();
+        }
     }
 
-    // set the new one
+    // set the new page
     m_page = page;
 
@@ -280,9 +310,8 @@
         // terminate successfully
         EndModal(wxID_OK);
-
         return TRUE;
     }
 
-    // send the event to the new page now
+    // send the change event to the new page now
     wxWizardEvent event(wxEVT_WIZARD_PAGE_CHANGED, GetId(), goingForward);
     (void)m_page->GetEventHandler()->ProcessEvent(event);
@@ -293,7 +322,17 @@
     m_page->Show();
 
-    // change the bitmap if necessary (and if we have it at all)
-    int bmpIsDefault = !m_page->GetBitmap().Ok();
-    if ( m_statbmp && (bmpIsDefault != bmpWasDefault) )
+    // check if bitmap needs to be updated
+    // update default flag as well
+    if(m_page->GetBitmap().Ok())
+    {
+        ThisBitmap = m_page->GetBitmap();
+        bmpIsDefault = FALSE;
+    }
+
+    // change the bitmap if:
+    // 1) a default bitmap was selected in constructor
+    // 2) this page was constructed with a bitmap
+    // 3) this bitmap is not the previous bitmap
+    if( m_statbmp && (ThisBitmap != PreviousBitmap) )
     {
         wxBitmap bmp;

 	  	 
